### PR TITLE
README: add cross-compile

### PR DESCRIPTION
**Pull request by zhangfeiv0**
Currently, the SLEEF documentation lacks detailed instructions on how to perform cross-compilation, which leads users to search for related issues or CI build processes to solve problems when they encounter errors. Here are some issues encountered during cross-compilation:
- [Unable to crosscompile without first building natively · Issue #249 · shibatch/sleef](https://github.com/shibatch/sleef/issues/249)
- [No rule to make target '/bin/mkrename' (and similar) when cross-compiling for Android · Issue #308 · shibatch/sleef](https://github.com/shibatch/sleef/issues/308)
- [No rule to make target '/bin/mkdisp' crosscompile using sleef/toolchains/aarch64-gcc.cmake · Issue #557 · shibatch/sleef](https://github.com/shibatch/sleef/issues/557)
- [Failed to cross compile: '/bin/mkalias', needed by 'sleef/src/libm/include/alias_advsimd.h', missing and no known rule to make it · Issue #51710 · pytorch/pytorch](https://github.com/pytorch/pytorch/issues/51710)

Therefore, I have added instructions on how to perform cross-compilation in the README. The first method involves incorporating the SLEEF binaries that have been built natively; the second method is similar to the current SLEEF CI testing process of "Setup QEMU," and by setting the environment variable `SLEEF_TARGET_EXEC_USE_QEMU`, it allows for the compilation and execution of files like mkrename and mkalias for the target architecture without first compiling a native SLEEF. This approach is taken because SLEEF is often integrated as part of other software, such as pytorch/third_party/sleef. By introducing this method, it can reduce the need to provide a native SLEEF directory for these software packages.

### CODE REVIEW

**Comment by blapie** [comment]
Hello! Thank you very much for this initiative!

I would be keen to move this to the actual documentation and keep the main README as light as possible.

There is a section on cross-compilation [here](https://github.com/shibatch/sleef/tree/master/docs/1-user-guide#cross-compilation-for-ios-and-android).

Maybe we could add a section before that specifically for Linux and point to it from the main README?

The changes look fine though, I'll just make a few comments, let's just make sure to not repeat information that is not already there.

Cheers!

File: README.md [hunk 1]
```diff
@@ -190,6 +190,51 @@ ctest --test-dir build -j
 
 For more detailed build instructions please refer to the [dedicated section on CMake](./docs/1-user-guide/build-with-cmake) or to [our web page][build_info_url].
 
+## How to CROSS-COMPILE SLEEF
+Two methods are used for cross-compiling SLEEF. Here are examples of cross-compiling SLEEF for the arm architecture on a platform with X86 architecture and Linux OS:
```
Comment (blapie) [inline README.md:194]: Maybe add that both rely on existing toolchain files provided in the `toolchains/` directory.

File: README.md [hunk 2]
```diff
@@ -190,6 +190,51 @@ ctest --test-dir build -j
...
+2. Cross-compile the target platform's SLEEF.
+```bash
+cmake -DCMAKE_TOOLCHAIN_FILE=./toolchains/aarch64-gcc.cmake -DNATIVE_BUILD_DIR=$(pwd)/build-native/ -S . -B build
+
+cmake --build build -j --clean-first
+```
+
+### Method 2
+
+No need to compile the native SLEEF
```
Comment (blapie) [inline README.md:213]: Maybe add "If running via an emulator like QEMU there is ..."

File: README.md (hunk 1 above)
Comment (blapie) [inline README.md:194]: AArch64

Comment (blapie) [inline README.md:194]: say x86_64

**Comment by zhangfeiv0** [comment]
Hi, @blapie 

I have made the changes according to your suggestions, could you please review it again?

File: docs/1-user-guide/README.md [hunk 3]
```diff
@@ -204,6 +205,62 @@ target_link_libraries(hellox86 sleef)
...
 
+<h2 id="cross_linux">Cross compilation for Linux</h2>
+
+Two methods are used for cross-compiling SLEEF. Both rely on existing toolchain
+files provided in the `toolchains/` directory.
+
+Here are examples of cross-compiling SLEEF for the AArch64 on a platform with 
+x86_64 and Linux OS:
+
+<h3 id="method1">Method1</h3>
```
Comment (blapie) [inline docs/1-user-guide/README.md:216]: Method 1

File: docs/1-user-guide/README.md [hunk 4]
```diff
@@ -204,6 +205,62 @@ target_link_libraries(hellox86 sleef)
...
+```
+
+2. Cross-compile the target platform's SLEEF.
+```bash
+cmake -DCMAKE_TOOLCHAIN_FILE=./toolchains/aarch64-gcc.cmake -DNATIVE_BUILD_DIR=$(pwd)/build-native/ -S . -B build
+
+cmake --build build -j --clean-first
+```
+
+<h3 id="method2">Method2</h3>
```
Comment (blapie) [inline docs/1-user-guide/README.md:234]: Method 2

**Review by blapie — COMMENTED** [review]
Looks great thanks! Just left some nitty comments.

You don't have to but if you could maybe change the commands in the iOS section to match yours (ie use cmake), that would be great:

```
# Build natively first
cmake -S . -B build-native
cmake --build build-native -j --clean-first
# Then cross-compile for iOS 
cmake  -S . -B build-cross -DCMAKE_TOOLCHAIN_FILE=./toolchains/ios.toolchain.cmake -DNATIVE_BUILD_DIR=$(pwd)/build-native -DSLEEF_DISABLE_MPFR=TRUE -DSLEEF_DISABLE_SSL=TRUE
```
and
```
# Build natively first
cmake -S . -B build-native
cmake --build build-native -j --clean-first
# Then cross-compile for Android 
cmake  -S . -B build-cross -DCMAKE_TOOLCHAIN_FILE=/opt/android-ndk-r21d/build/cmake/android.toolchain.cmake -DNATIVE_BUILD_DIR=$(pwd)/build-native -DANDROID_ABI=arm64-v8a
```

**Comment by zhangfeiv0** [comment]
@blapie  The changes have been made, please review.

**Review by blapie — APPROVED** [review]

**Pull request merged**